src/main/java/teammates/logic/core/InstructorsLogic.java

```diff
@@ -174,6 +174,18 @@ public void verifyIsEmailOfInstructorOfCourse(String instructorEmail, String cou
         }
     }
 
+    public void verifyAtLeastOneInstructorIsDisplayed(boolean isOriginalInstructorDisplayed, String courseId,
+                                                      boolean isEditedInstructorDisplayed)
+            throws InvalidParametersException {
+        List<InstructorAttributes> instructorsDisplayed = instructorsDb.getInstructorsDisplayedToStudents(courseId);
+        boolean isEditedInstructorChangedToNonVisible = isOriginalInstructorDisplayed && !isEditedInstructorDisplayed;
+
+        if (!(instructorsDisplayed.isEmpty() && !isOriginalInstructorDisplayed && isEditedInstructorDisplayed)
```

Comment:
Denoting condition `instructorsDisplayed.isEmpty() && !isOriginalInstructorDisplayed && isEditedInstructorDisplayed` as `x` and condition `instructorsDisplayed.size() == 1 && isEditedInstructorChangedToNonVisible` as `y`, what we want is the exception to be thrown whenever `!x || y`, but your condition here is `!x && y`.

Reply:
For condition x, if `instructorsDisplayed.isEmpty()` is false, meaning there is atleast one instructor is displayed, then x will evaluate to false and !x will be true, which means an exception will be thrown?

Reply:
Maybe a better condition is `x || y` where `y` is identical to previous and `x` is `instructorsDisplayed.isEmpty() && !isEditedInstructorDisplayed`.

Reply:
If condition x evaluates to be true, we do not want the exception to be thrown? Is that right?

Reply:
No; if either is true, an exception should be thrown.

Reply:
> Maybe a better condition is `x || y` where `y` is identical to previous and `x` is `instructorsDisplayed.isEmpty() && !isEditedInstructorDisplayed`.

Yeah this seems good